### PR TITLE
feat: updating remark directives for new docs

### PR DESCRIPTION
**Pull request by elylucas**

### CODE REVIEW

File: src/components/cypress-config-file-tabs/index.tsx [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  children: React.ReactNode;
+}
+
+const CypressConfigFileTabs: React.FC<CypressConfigFileTabsProps> = ({ children }) => {
+  return (
+    <Tabs groupId="config-file-tabs">
+      <TabItem value="js" label="cypress.config.js">
+        {children && children[0]}
+      </TabItem>
+      <TabItem value="ts" label="cyppress.config.ts">
```
Comment (jaffrepaul) [inline src/components/cypress-config-file-tabs/index.tsx:15]:
```suggestion
      <TabItem value="ts" label="cypress.config.ts">
```

File: docs/guides/references/cypress-studio.mdx [hunk 2]
```diff
@@ -7,7 +7,7 @@ title: Cypress Studio
 <strong class="alert-header"><Icon name="exclamation-triangle"></Icon>
 Removed</strong>
 
-Cypress Studio has been removed in Cypress v10 and will be rethought/revisited
+Cypress Studio has been removed in Cypress 10 and will be rethought/revisited
```
Comment (jaffrepaul) [inline docs/guides/references/cypress-studio.mdx:10]: 😆 this was a hot topic at one point and we made all refs "v10". Did that change again?

File: docs/guides/references/migration-guide.mdx [hunk 3]
```diff
@@ -56,17 +56,29 @@ Related notes:
   configuration using `cypress.config.js` and `cypress.config.ts` vs the older
   `cypress.json`. For example:
 
-:::cypress-config-example
+<CypressConfigFileTabs>
```
Comment (jaffrepaul) [inline docs/guides/references/migration-guide.mdx:59]: This is another example that in prod shows a deprecated 3rd tab with the .json version.

**Review by jaffrepaul — COMMENTED** [review]
Added a few questions and one typo for the .ts tab. 

Side note: The codeblocks in the light view get a little lost without any sort of surrounding border. But I suppose that's what light mode would offer. We can also consider making the code blocks dark in both themes, which gives a natural contrast in these elements.

File: src/components/cypress-config-file-tabs/index.tsx (hunk 1 above)
Comment (elylucas) [reply src/components/cypress-config-file-tabs/index.tsx:15]: sometimes i wonder about myself

File: docs/api/cypress-api/env.mdx [hunk 4]
```diff
@@ -141,18 +168,31 @@ Cypress.env('host') // => http://server.dev.local
 
 #### Override multiple values from the Cypress configuration by passing an object
 
-:::cypress-config-example
+<CypressConfigFileTabs>
```
[the remark this answers is not in the document]
Comment (elylucas) [reply docs/api/cypress-api/env.mdx:171]: Ya, this was discussed and it came down to the docs should reflect the current state of the app, not a past state. The old docs will be available for those that need to look up old config.

File: docs/guides/references/cypress-studio.mdx (hunk 2 above)
Comment (elylucas) [reply docs/guides/references/cypress-studio.mdx:10]: ya dont think we are suppose to use 'v' in front of the versions. I  asked Matt V for clarificaiton.

File: docs/guides/references/migration-guide.mdx [hunk 5]
```diff
@@ -100,9 +112,9 @@ Related notes:
 
 ```js
 // bind to the event we care about
-on("<event>", (arg1, arg2) => {
+on('<event>', (arg1, arg2) => {
```
[the remark this answers is not in the document]
Comment (elylucas) [reply docs/guides/references/migration-guide.mdx:115]: there are lint rules that format code blocks in the project, so i assume so?

File: plugins/cypressRemarkPlugins/tsconfig.json [hunk 6]
```diff
@@ -0,0 +1,105 @@
+{
```
Comment (jaffrepaul) [inline plugins/cypressRemarkPlugins/tsconfig.json:1]: wow, what's this all about?

File: docs/api/commands/fixture.mdx [hunk 7]
```diff
@@ -183,33 +183,21 @@ cy.fixture('users').then((json) => {
...
+:::visit-mount-example
 
 ```js
 cy.fixture('user').then((user) => {
   user.firstName = 'Jane'
   cy.intercept('GET', '/users/1', user).as('getUser')
 })
 
-cy.visit('/users')
+-{cy.visit('/users')::}-cy.mount(<Users />)}-
```
Comment (jaffrepaul) [inline docs/api/commands/fixture.mdx:194]: some render wonk on this one

Screen Shot 2022-07-11 at 5 07 28 PM

File: docs/api/commands/request.mdx [hunk 8]
```diff
@@ -371,20 +371,29 @@ actual, running server without having to start the front end application.
 
 ### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
```
Comment (jaffrepaul) [inline docs/api/commands/request.mdx:373]: Looks like we lost the code formatting on these list items.

File: plugins/cypressRemarkPlugins/src/cypressConfigPluginExample/readme.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+code block exists, it will be included in the `setupNodeEvents` method.
+
+It uses the `CypressConfigFileTabs` component to display each code block in its
+own tab.
+
+Usage:
+
+:::cypress-config-plugin-example
+
+```ts
```
Comment (jaffrepaul) [inline plugins/cypressRemarkPlugins/src/cypressConfigPluginExample/readme.md:17]: is there a difference in tagging these "ts" vs "typescript"? 

Can they be consistent if there isn't a functional difference? Just less to mentally parse.

File: docs/api/commands/viewport.mdx [hunk 10]
```diff
@@ -298,15 +279,21 @@ describe(
 
 ### Requirements [<Icon name="question-circle"/>](/guides/core-concepts/introduction-to-cypress#Chains-of-Commands)
 
-<List><li>`cy.viewport()` requires being chained off of `cy`.</li></List>
+<List>
```
Comment (jaffrepaul) [inline docs/api/commands/viewport.mdx:282]: Same issue here on the list stylez

File: plugins/cypressRemarkPlugins/.vscode/launch.json [hunk 11]
```diff
@@ -0,0 +1,18 @@
+{
```
Comment (jaffrepaul) [inline plugins/cypressRemarkPlugins/.vscode/launch.json:1]: What does this get us?

**Review by jaffrepaul — COMMENTED** [review]
Added some questions & comments. The list/code styling didn't originate in this PR but noting it anyway.

File: plugins/cypressRemarkPlugins/src/cypressConfigPluginExample/readme.md (hunk 9 above)
Comment (elylucas) [reply plugins/cypressRemarkPlugins/src/cypressConfigPluginExample/readme.md:17]: Theres not. Doesn't hurt but its good to be consistent.

Comment (elylucas) [reply plugins/cypressRemarkPlugins/src/cypressConfigPluginExample/readme.md:17]: Same with js vs javascript

File: plugins/cypressRemarkPlugins/.vscode/launch.json (hunk 11 above)
Comment (elylucas) [reply plugins/cypressRemarkPlugins/.vscode/launch.json:1]: ahh, probably shouldn't of been checked in. This was before I moved the plugins to be their own package and I set this up to debug the ts code.

Comment (elylucas) [reply plugins/cypressRemarkPlugins/.vscode/launch.json]: i removed it

File: package.json [hunk 12]
```diff
@@ -5,11 +5,13 @@
   "scripts": {
     "docusaurus": "docusaurus",
     "start": "docusaurus start",
-    "prebuild": "rm -rf dist",
+    "prebuild": "rm -rf dist && yarn build:plugins",
```
Comment (jaffrepaul) [inline package.json:8]: would be great to cover these scripts when we run through all this as a team

**Review by jaffrepaul — APPROVED** [review]
LGTM. Thanks for taking all this on!

**Pull request merged**